test-array.cpp

```diff
@@ -60,18 +60,14 @@ void test_array_add() {
     Object *second = arr->get(1);
     t_true(second->equals(u));
 
-    Object *third = arr->get(2);
+    Object *third = arr->get(3);
```

Comment:
I'm a bit confused as to why you are trying to get at the index 3, since there are not 3 elements in the list? I think our test is poorly written. Would it make sense to you if we changed it to be add(s, 0) then add(t, 1), then add(u,2) so that elements are added in sequential order? This would maintain the java arraylist property

Reply:
I think shifting the elements is the proper implementation, but our test didn't show this.
It should be:
add("hi", 0)
add("a", 1)
add("b", 1)
makes ["hi", "b", "a"]

Reply:
Does that make sense

Reply:
So if the tests / behavior need to change that's ok, but currently there are 4 elements because of the padding.  When add is called with an index greater than size, we pad up to the point. When we then add in the middle, it shifts elements. 

If you want to only allow adding in order, that should be a documentation and a test change

Reply:
Ah ok, we gotcha. Looks like it makes sense and fits the requirements. Going to merge it.